test/product-import.spec.coffee

```diff
@@ -470,3 +470,137 @@ describe 'ProductImport', ->
       updatedProduct = @import._ensureDefaults(sampleProduct)
       expect(updatedProduct).toEqual expectedProduct
 
+  describe ':: Custom reference resolution', ->
+
+    it ' :: should detect reference type attribute', ->
+      sampleReferenceAttribute =
+        name: 'foobar'
+        value:
+          value: 'some value'
+          resolvePredicate: 'some predicate query'
+          endpoint: 'some endpoint'
```

Comment:
Where does in real life `resolvePredicate` and `endpoint` are coming from?
If you store those calculation data/logic within the actually data, you might emphasis is better.
Maybe `_endpoint` or put both keys inside of `_resolveHelper`